### PR TITLE
Add information about dealing with Jar Hell from IntelliJ

### PR DESCRIPTION
**Pull request by dadoonet**

### CODE REVIEW

File: TESTING.asciidoc [hunk 1]
```diff
@@ -537,6 +537,16 @@ ES_JAVA_OPTS="-Xdebug -Xrunjdwp:server=y,transport=dt_socket,address=4000,suspen
...
 Read your IDE documentation for how to attach a debugger to a JVM process.
 
+== Running unit tests from IntelliJ
+
+If you try to run the unit tests from IntelliJ, you might suffer from a Jar Hell issue.
+
+To fix it, add `idea.no.launcher=true` to your `idea.properties` and restart IntelliJ:
+
+---------------------------------------------------------------------------
+echo idea.no.launcher=true >> ~/Library/Preferences/IntelliJIdea2017.1/idea.properties
```
Comment (javanna) [inline TESTING.asciidoc]: wasn't it idea.vmoptions ?

Comment (jasontedor) [inline TESTING.asciidoc]: I don't think we should tie these instructions to a specific IntelliJ version. There's a way to do this *within* IntelliJ by going to `Help -> Custom VM options` and adding the system property there. I think we should encourage that as then it's version independent and has the same impact.

Comment (dadoonet) [inline TESTING.asciidoc]: @javanna Apparently on 2017.2 I had to add it to `idea.properties` although it was there with `-Didea.no.launcher=true` in `idea.vmoptions`.

@jasontedor fair point. Updated but with Custom properties instead.

LMK guys.

**Comment by hub-cap** [comment]
Should there be a mention of this in Contributing guide as well, since its setup related?

**Comment by jasontedor** [comment]
@hub-cap Good one, I think this should be moved there where IDE setup is discussed, separated from testing all together.

**Comment by dadoonet** [comment]
Well. It's already in the CONTRIBUTING guide: https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-to-the-elasticsearch-codebase

So I'm closing this one.

**Pull request closed without merging**